File: classifier/data/raw/ai/samples/project_euler_java/Problem010.java
/** Project Euler Problem 10: sum of all primes below two million. */
public class Problem010 {
    public static void main(String[] args) {
        final int limit = 2_000_000;
        boolean[] composite = new boolean[limit];
        long sum = 0;
        for (int n = 2; n < limit; n++) {
            if (!composite[n]) {
                sum += n;
                if ((long) n * n < limit) {
                    for (int multiple = n * n; multiple < limit; multiple += n) {
                        composite[multiple] = true;
                    }
                }
            }
        }
        System.out.println(sum);
    }
}

File: classifier/data/raw/ai/samples/project_euler_zig/problem_010.zig
const std = @import("std");

// Project Euler Problem 10: sum all primes below two million using a sieve.
pub fn main() !void {
    const limit = 2_000_000;
    var composite: [limit]bool = [_]bool{false} ** limit;

    var prime: usize = 2;
    while (prime <= (limit - 1) / prime) : (prime += 1) {
        if (composite[prime]) continue;
        var multiple = prime * prime;
        while (multiple < limit) : (multiple += prime) {
            composite[multiple] = true;
        }
    }

    var sum: u64 = 0;
    var n: usize = 2;
    while (n < limit) : (n += 1) {
        if (!composite[n]) sum += n;
    }
    try printNumber(sum);
}

fn printNumber(number: anytype) !void {
    var buffer: [32]u8 = undefined;
    const output = try std.fmt.bufPrint(&buffer, "{d}\n", .{number});
    try std.Io.File.writeStreamingAll(.stdout(), std.Io.Threaded.global_single_threaded.io(), output);
}
